src/nvs.rs

```diff
@@ -352,6 +352,305 @@ impl<T: NvsPartitionId> EspNvs<T> {
 
         Ok(true)
     }
+
+    pub fn get_blob<'a>(
+        &self,
+        name: &str,
+        buf: &'a mut [u8],
+    ) -> Result<Option<&'a [u8]>, EspError> {
+        let c_key = CString::new(name).unwrap();
+
+        // check for blob value, by getting blob length
```

Comment:
I don't think you need to call `nvs_get_blob` twice. Perhaps you copied from the current Rust-only impl? Just follow the same logic as in `get_str`.